sources/api/host-containers/src/main.rs

```diff
@@ -434,22 +434,48 @@ where
     Ok(())
 }
 
+fn is_restart_required(changed_settings: &str, container_name: &str) -> bool {
+    if changed_settings.is_empty() {
+        // it means that Bottlerocket is booting - all containers need to be started
+        return true;
+    }
+
+    let settings: Vec<&str> = changed_settings.split_whitespace().collect();
+    let setting_prefix = format!("host-containers.{}", container_name);
+
+    for setting in settings.iter() {
+        if setting.contains(&setting_prefix) {
```

Comment:
I think we could be a little more careful here to match the string we know will be present.  Something like:
```rust
     let setting_prefix = "settings.host-containers.";
     let container_prefix = format!("{}.{}.", setting_prefix, container_name);
...
     if setting.starts_with(&container_prefix) {
```
(I'm suggesting two variables so I can use `setting_prefix` in a later comment...)